File: app/src/main/java/com/radio/app/network/RetrofitClient.java
package com.radio.app.network;

public class RetrofitClient {
    private static RetrofitClient instance;

    private RetrofitClient() {}

    public static synchronized RetrofitClient getInstance() {
        if (instance == null) instance = new RetrofitClient();
        return instance;
    }

    public RadioApiService getApiService() {
        return RadioApiService.getInstance();
    }

    public EpisodeApiService getEpisodeApiService() {
        return EpisodeApiService.getInstance();
    }
}

File: app/src/main/java/com/radio/app/network/RetrofitClient.kt
package com.radio.app.network

class RetrofitClient private constructor() {

    companion object {
        private var instance: RetrofitClient? = null

        @Synchronized
        fun getInstance(): RetrofitClient {
            return instance ?: RetrofitClient().also { instance = it }
        }
    }

    fun getApiService(): RadioApiService = RadioApiService.getInstance()

    fun getEpisodeApiService(): EpisodeApiService = EpisodeApiService.getInstance()
}
